From ac38f280fef891912c22cdac44ac60d85e94f3b9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Tue, 11 Oct 2016 13:30:39 -0400
Subject: [PATCH 06/13] Disable ctor COMDATs on Solaris

---
 lib/CodeGen/ItaniumCXXABI.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/CodeGen/ItaniumCXXABI.cpp b/lib/CodeGen/ItaniumCXXABI.cpp
index 490fadbd1c..fa3772b948 100644
--- a/lib/CodeGen/ItaniumCXXABI.cpp
+++ b/lib/CodeGen/ItaniumCXXABI.cpp
@@ -3535,6 +3535,10 @@ static StructorCodegen getCodegenToUse(CodeGenModule &CGM,
     return StructorCodegen::RAUW;
 
   if (llvm::GlobalValue::isWeakForLinker(Linkage)) {
+    // COMDATs with arbitrary names break Solaris
+    if (CGM.getTarget().getTriple().getOS() == llvm::Triple::Solaris)
+      return StructorCodegen::Emit;
+
     // Only ELF and wasm support COMDATs with arbitrary names (C5/D5).
     if (CGM.getTarget().getTriple().isOSBinFormatELF() ||
         CGM.getTarget().getTriple().isOSBinFormatWasm())
-- 
2.11.0

