From fb55bf80819d7103ea6939594f7c3444fc2ee540 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Tue, 11 Oct 2016 17:18:03 -0400
Subject: [PATCH 08/13] Enable compiler-rt support on Solaris

---
 lib/Driver/Tools.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/lib/Driver/Tools.cpp b/lib/Driver/Tools.cpp
index 92aed8a05c..52d8fefc39 100644
--- a/lib/Driver/Tools.cpp
+++ b/lib/Driver/Tools.cpp
@@ -53,6 +53,9 @@ using namespace clang::driver::tools;
 using namespace clang;
 using namespace llvm::opt;
 
+static void AddRunTimeLibs(const ToolChain &TC, const Driver &D,
+                           ArgStringList &CmdArgs, const ArgList &Args);
+
 static void handleTargetFeaturesGroup(const ArgList &Args,
                                       std::vector<StringRef> &Features,
                                       OptSpecifier Group) {
@@ -8918,10 +8921,9 @@ void solaris::Linker::ConstructJob(Compilation &C, const JobAction &JA,
   if (!Args.hasArg(options::OPT_nostdlib, options::OPT_nodefaultlibs)) {
     if (getToolChain().getDriver().CCCIsCXX())
       getToolChain().AddCXXStdlibLibArgs(Args, CmdArgs);
-    CmdArgs.push_back("-lgcc_s");
+    AddRunTimeLibs(getToolChain(), getToolChain().getDriver(), CmdArgs, Args);
     CmdArgs.push_back("-lc");
     if (!Args.hasArg(options::OPT_shared)) {
-      CmdArgs.push_back("-lgcc");
       CmdArgs.push_back("-lm");
     }
   }
@@ -10120,6 +10122,7 @@ static void AddRunTimeLibs(const ToolChain &TC, const Driver &D,
     case llvm::Triple::Win32:
     case llvm::Triple::Linux:
     case llvm::Triple::Fuchsia:
+    case llvm::Triple::Solaris:
       CmdArgs.push_back(TC.getCompilerRTArgString(Args, "builtins"));
       break;
     }
-- 
2.11.1

