From a50ebfd8468936f8ba84ffae6659c2ad5d282c4d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Sun, 16 Oct 2016 19:54:13 +0200
Subject: [PATCH 11/13] [Driver] Support overriding linker via LDEXEC

---
 lib/Driver/ToolChain.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/Driver/ToolChain.cpp b/lib/Driver/ToolChain.cpp
index 6776a36..2de3680 100644
--- a/lib/Driver/ToolChain.cpp
+++ b/lib/Driver/ToolChain.cpp
@@ -378,6 +378,10 @@ std::string ToolChain::GetLinkerPath() const {
 
     getDriver().Diag(diag::err_drv_invalid_linker_name) << A->getAsString(Args);
     return "";
+  } else {
+    const char* ldexec = getenv("LDEXEC");
+    if (ldexec && *ldexec)
+      return ldexec;
   }
 
   return GetProgramPath(DefaultLinker);
-- 
2.10.1

