From 6a73f28a51b658a6035d0d9c766441ae9e454100 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Tue, 25 Oct 2016 21:32:47 +0200
Subject: [PATCH 14/14] Pass -leh for sanitizer

---
 lib/Driver/Tools.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/Driver/Tools.cpp b/lib/Driver/Tools.cpp
index 6f77418bef..4826609ea3 100644
--- a/lib/Driver/Tools.cpp
+++ b/lib/Driver/Tools.cpp
@@ -3309,6 +3309,9 @@ static bool addSanitizerRuntimes(const ToolChain &TC, const ArgList &Args,
   if (SanArgs.hasCrossDsoCfi() && !AddExportDynamic)
     CmdArgs.push_back("-export-dynamic-symbol=__cfi_check");
 
+  if (!StaticRuntimes.empty())
+    CmdArgs.push_back("-leh");
+
   return !StaticRuntimes.empty();
 }
 
-- 
2.11.0.rc2

