From d88c874f8f7bfef237ec53b392744a2500e784c2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Tue, 25 Oct 2016 21:32:47 +0200
Subject: [PATCH 12/12] Pass -leh for sanitizer

---
 lib/Driver/Tools.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/Driver/Tools.cpp b/lib/Driver/Tools.cpp
index 009757e25e..b99ac93345 100644
--- a/lib/Driver/Tools.cpp
+++ b/lib/Driver/Tools.cpp
@@ -3422,6 +3422,9 @@ static bool addSanitizerRuntimes(const ToolChain &TC, const ArgList &Args,
   if (SanArgs.hasCrossDsoCfi() && !AddExportDynamic)
     CmdArgs.push_back("-export-dynamic-symbol=__cfi_check");
 
+  if (!StaticRuntimes.empty() || !NonWholeStaticRuntimes.empty())
+    CmdArgs.push_back("-leh");
+
   return !StaticRuntimes.empty() || !NonWholeStaticRuntimes.empty();
 }
 
-- 
2.11.1

