From bf3565e51a05adce08b23cdc2a8fd1972844ac0b Mon Sep 17 00:00:00 2001
From: miguelzf <eligmu@gmail.com>
Date: Sun, 2 Oct 2016 08:17:27 -0400
Subject: [PATCH 1/2] Fix wrapped loop iterator in PromoteMem2Reg

---
 lib/Transforms/Utils/PromoteMemoryToRegister.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/Transforms/Utils/PromoteMemoryToRegister.cpp b/lib/Transforms/Utils/PromoteMemoryToRegister.cpp
index 35faa6f..46dbfeb 100644
--- a/lib/Transforms/Utils/PromoteMemoryToRegister.cpp
+++ b/lib/Transforms/Utils/PromoteMemoryToRegister.cpp
@@ -277,7 +277,7 @@ public:
   void run();
 
 private:
-  void RemoveFromAllocasList(unsigned &AllocaIdx) {
+  void RemoveFromAllocasList(signed &AllocaIdx) {
     Allocas[AllocaIdx] = Allocas.back();
     Allocas.pop_back();
     --AllocaIdx;
@@ -525,7 +525,7 @@ void PromoteMem2Reg::run() {
   LargeBlockInfo LBI;
   ForwardIDFCalculator IDF(DT);
 
-  for (unsigned AllocaNum = 0; AllocaNum != Allocas.size(); ++AllocaNum) {
+  for (signed AllocaNum = 0; AllocaNum != Allocas.size(); ++AllocaNum) {
     AllocaInst *AI = Allocas[AllocaNum];
 
     assert(isAllocaPromotable(AI) && "Cannot promote non-promotable alloca!");
-- 
2.10.1

