From 2838585905e3f84c56c48097c5acc3c4a573ece5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Tue, 11 Oct 2016 17:30:27 -0400
Subject: [PATCH 2/2] [Orc] Fix templates for Solaris

---
 include/llvm/ExecutionEngine/Orc/RPCByteChannel.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/llvm/ExecutionEngine/Orc/RPCByteChannel.h b/include/llvm/ExecutionEngine/Orc/RPCByteChannel.h
index c8cb42d..1bf8ea4 100644
--- a/include/llvm/ExecutionEngine/Orc/RPCByteChannel.h
+++ b/include/llvm/ExecutionEngine/Orc/RPCByteChannel.h
@@ -144,14 +144,14 @@ public:
 };
 
 template <typename ChannelT>
-class SerializationTraits<ChannelT, uint8_t>
+class SerializationTraits<ChannelT, unsigned char>
   : public RPCByteChannelPrimitiveSerialization<ChannelT, uint8_t> {
 public:
   static const char* getName() { return "uint8_t"; }
 };
 
 template <typename ChannelT>
-class SerializationTraits<ChannelT, int8_t>
+class SerializationTraits<ChannelT, signed char>
   : public RPCByteChannelPrimitiveSerialization<ChannelT, int8_t> {
 public:
   static const char* getName() { return "int8_t"; }
-- 
2.10.1

